ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/AssertTopicExecutor.java

```diff
@@ -0,0 +1,122 @@
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.server.execution;
+
+import static io.confluent.ksql.rest.Errors.assertionFailedError;
+import static io.confluent.ksql.util.KsqlConfig.KSQL_ASSERT_TOPIC_DEFAULT_TIMEOUT_MS;
+
+import io.confluent.ksql.KsqlExecutionContext;
+import io.confluent.ksql.execution.expression.tree.Literal;
+import io.confluent.ksql.parser.tree.AssertTopic;
+import io.confluent.ksql.rest.SessionProperties;
+import io.confluent.ksql.rest.entity.AssertTopicEntity;
+import io.confluent.ksql.rest.server.resources.KsqlRestException;
+import io.confluent.ksql.services.KafkaTopicClient;
+import io.confluent.ksql.services.ServiceContext;
+import io.confluent.ksql.statement.ConfiguredStatement;
+import io.confluent.ksql.util.KsqlException;
+import io.confluent.ksql.util.RetryUtil;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public final class AssertTopicExecutor {
+
+  private static final int RETRY_MS = 100;
+
+  private AssertTopicExecutor() {
+
+  }
+
+  public static StatementExecutorResponse execute(
+      final ConfiguredStatement<AssertTopic> statement,
+      final SessionProperties sessionProperties,
+      final KsqlExecutionContext executionContext,
+      final ServiceContext serviceContext
+  ) {
+    final KafkaTopicClient client = serviceContext.getTopicClient();
+    final int timeout = statement.getStatement().getTimeout().isPresent()
+        ? (int) statement.getStatement().getTimeout().get().toDuration().toMillis()
+        : executionContext.getKsqlConfig().getInt(KSQL_ASSERT_TOPIC_DEFAULT_TIMEOUT_MS);
+    try {
+      RetryUtil.retryWithBackoff(
+          timeout / RETRY_MS,
+          RETRY_MS,
+          RETRY_MS,
+          () -> assertTopic(
+              client, statement.getStatement().getTopic(), statement.getStatement().getConfig())
+      );
+    } catch (final KsqlException e) {
+      throw new KsqlRestException(assertionFailedError(e.getMessage()));
+    }
+    return StatementExecutorResponse.handled(Optional.of(
+        new AssertTopicEntity(statement.getStatementText(), statement.getStatement().getTopic())));
+  }
+
+  private static void assertTopic(
+      final KafkaTopicClient client,
+      final String topic,
+      final Map<String, Literal> config
+  ) {
+    final boolean topicExists;
+    try {
+      topicExists = client.isTopicExists(topic);
+    } catch (final Exception e) {
+      throw new KsqlException("Cannot check that topic exists: " + e.getMessage());
+    }
+
+    if (topicExists) {
+      final int partitions = client.describeTopic(topic).partitions().size();
+      final int replicas = client.describeTopic(topic).partitions().get(0).replicas().size();
+      final List<String> configErrors = new ArrayList<>();
+      config.forEach((k, v) -> {
```

Comment:
As a non-blocking comment, the part of me that's an Scala/FP junkie would offer that it may be an interesting exercise to see if it is easy to switch to a "map" (or maybe a flatmap) and collect the result for the errors.  

This code works and is clear, so no obligation to try that.

Reply:
Leaving it as is because I tried that end it ended up getting messier :(